tools/generator/templates/player/python/interface.cc.jinja2

```diff
@@ -0,0 +1,363 @@
+    return tuple;
+    {% else %}
+    PyObject* name = PyUnicode_FromString("{{ struct.str_name }}");
+    PyObject* cstr = PyObject_GetAttr(py_module, name);
+    Py_DECREF(name);
+    if (cstr == NULL)
+        throw 42;
+    PyObject* ret = PyObject_CallObject(cstr, tuple);
+    Py_DECREF(cstr);
+    Py_DECREF(tuple);
+    if (ret == NULL)
+        throw 42;
+    return ret;
+    {% endif %}
+}
+
+template <>
+{{ struct.str_name }} python_to_cxx<PyObject*, {{ struct.str_name }}>(PyObject* in)
+{
+    {{ struct.str_name }} out;
+    PyObject* i;
+    {% for field_name, field_type, field_comment in struct.str_field %}
+
+    {{ field_comment|cxx_comment }}
+        {% if struct.str_tuple %}
+    i = PyTuple_GetItem(in, {{ loop.index0 }});
+        {% else %}
+    i = cxx_to_python<PyObject*, int>({{ loop.index0 }});
+    i = PyObject_GetItem(in, i);
+        {% endif %}
+    if (i == NULL)
+        throw 42;
+    out.{{ field_name }} = {{ field_type|python_to_cxx }}(i);
+        {% if not struct.str_tuple %}
+    Py_DECREF(i);
+        {% endif %}
+    {% endfor %}
+
+    return out;
+}
+{% endfor %}
+
+{% for func in game.function %}
+
+{{ func.fct_summary|cxx_comment }}
+static PyObject* p_{{ func.fct_name }}(PyObject* self, PyObject* args)
+{
+    (void) self;
+    {% for arg_name, _, _ in func.fct_arg %}
+    PyObject* arg_{{ arg_name }};
+    {% endfor %}
+    if (!PyArg_ParseTuple(args, "{{ 'O' * (func.fct_arg|length) }}"
+        {%- for arg_name, _, _ in func.fct_arg -%}
+        , &arg_{{ arg_name }}
+        {%- endfor %}))
+    {
+        return NULL;
+    }
+
+    try {
+        {%+ if func is returning %}return {% endif -%}
+        {{ func.fct_ret_type|cxx_to_python }}(api_{{ func.fct_name }}(
+        {%- set comma = joiner(', ') -%}
+        {%- for arg_name, arg_type, _ in func.fct_arg -%}
+        {{ comma() }}{{ arg_type|python_to_cxx }}(arg_{{ arg_name }})
+        {%- endfor %}));
+        {% if not func is returning %}
+        Py_RETURN_NONE;
+        {% endif %}
+    } catch (...) {
+        return NULL;
+    }
+}
+{% endfor %}
+
+// API function to register.
+static PyMethodDef api_callback[] = {
+    {% for func in game.function %}
+    {"{{ func.fct_name }}", p_{{ func.fct_name }}, METH_VARARGS, "{{ func.fct_name }}"},
+    {% endfor %}
+    {NULL, NULL, 0, NULL}
+};
+
+// Initialize C module
+PyMODINIT_FUNC PyInit__api()
+{
+    static struct PyModuleDef apimoduledef = {
+        PyModuleDef_HEAD_INIT,
+        "_api",
+        "API module",
+        -1,
+        api_callback,
+        NULL,
+        NULL,
+        NULL,
+        NULL,
+    };
+    return PyModule_Create(&apimoduledef);
+}
+
+// Load a Python module
+static PyObject* _import_module(const char* m)
+{
+    PyObject* name = PyUnicode_FromString(m);
+    PyObject* mod = PyImport_Import(name);
+    Py_DECREF(name);
+    if (mod == NULL)
+        if (PyErr_Occurred())
+        {
+            PyErr_Print();
+            abort();
```

Comment:
(this is the kind of debugging message + termination I'm speaking about in other comments)